src/test/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImportIntegrationTest.java

```diff
@@ -198,34 +196,75 @@ private static void checkGenomicsDBAgainstExpected(final String workspace, final
         }
     }
 
-    @Test
-    public void testSampleMappingFileInsteadOfVCFs() throws IOException {
-        final File sampleNameFile = createTempSampleMapFile();
 
+    @DataProvider
+    public Iterator<Object[]> getOrderingTests(){
+        final File outOfOrderSampleMap = getSampleMapFile(
+                        "HG00268\t" + HG_00268 + "\n" +
+                        "NA19625\t" + NA_19625 + "\n" +
+                        "HG00096\t" + HG_00096);
+
+        final List<Integer> batchSizes = Arrays.asList(0, 1, 2, 3, 4);
+        final List<Object[]> results = new ArrayList<>();
+        for( final Integer batchSize: batchSizes){
+            // -V in order
+            results.add(new Object[] {new ArgumentsBuilder()
+                    .addArgument(GenomicsDBImport.BATCHSIZE_ARG_NAME, String.valueOf(batchSize))
+                    .addVCF(new File(HG_00096))
+                    .addVCF(new File(HG_00268))
+                    .addVCF(new File(NA_19625))});
+
+            // -V out of order
+            results.add(new Object[] {new ArgumentsBuilder()
+                    .addArgument(GenomicsDBImport.BATCHSIZE_ARG_NAME, String.valueOf(batchSize))
+                    .addVCF(new File(HG_00268))
+                    .addVCF(new File(NA_19625))
+                    .addVCF(new File(HG_00096))});
+
+            //in order sample map
+            results.add(new Object[] {new ArgumentsBuilder()
+                    .addArgument(GenomicsDBImport.BATCHSIZE_ARG_NAME, String.valueOf(batchSize))
+                    .addFileArgument(GenomicsDBImport.SAMPLE_NAME_MAP_LONG_NAME, createInOrderSampleMap())});
+
+            //out of order sample map
+            results.add(new Object[] {new ArgumentsBuilder()
+                    .addArgument(GenomicsDBImport.BATCHSIZE_ARG_NAME, String.valueOf(batchSize))
+                    .addFileArgument(GenomicsDBImport.SAMPLE_NAME_MAP_LONG_NAME, outOfOrderSampleMap)});
+        }
+        return results.iterator();
```

Comment:
Add a least one additional test case that uses `--vcfInitializerThreads` with a value > 1, since that is a different code path that you also modified.